### ISSUE TITLE
Multi-line tabline

### ISSUE THREAD

**Issue by jcfk** [vim/vim#9828]
**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**

Very sure it's impossible. Currently, when there are more tabs than the width of Vim, the tabline will be cut off on one side or the other so that you can't see all tabs at once.

**Describe the solution you'd like**

The option of enabling a multi-line tabline that wraps when it's too long, analogous to [this](https://657b072aab060d50f8ce-d7abb53cb376b4947d77643d4b4a48d3.ssl.cf1.rackcdn.com/27_94bd8862.png) idea in the browser.

**Describe alternatives you've considered**

Plenty of nice plugins exist for doing a similar thing in displaying buffers in a wrapping way, but of course this has nothing to do with tabs, so they aren't alternatives.

**Comment by chrisbra** [comment]
you are talking about the gui yes?

**Comment by jcfk** [comment]
Sorry, I should have specified. I use terminal vim, so the question is targeted at terminal vim.

**Comment by chrisbra** [comment]
But then you may loose a lot of space for each wrapped tabline. I am wondering if a popup window could help here?

I think we also have a issue here about a multi-line statusline, which is kind of a duplicate.

**Comment by jcfk** [comment]
Yes, each wrap would take one line from the window below. Presumably only users with enough space would enable it. But the nice thing about the tabline is that it's always visible.

**Comment by brammool** [comment]
A wrapping tabline is frowned upon from a UI perspective.  It breaks the connection with the physical equivalent.  It raises the problem that when selecting a tab in the upper row the connection with the page is missing.  Or the tab would move position, which is even worse.

The best alternative is to make a vertical, narrow column to place the tabs headers in.  Then there is plenty of space, and it's even possible to make it scroll for even more.  Not sure how much work it would be to implement.  Might even be possible with a plugin (but might not look perfectly).

**Comment by k-takata** [comment]
> The best alternative is to make a vertical, narrow column to place the tabs headers in.

@rbtnn's TabSideBar is available here: https://github.com/rbtnn/vim

**Comment by sankl** [comment]
TabSideBar is a great enhancement! Could this be taken into the main branch at some point?
This is an enabler to be put into .vimrc: https://github.com/rbtnn/vimrc/blob/master/vim/settings/tabsidebar.vim
